### ISSUE TITLE
Unable to load ZWave product file: 'kwikset/kwikset-9xx.xml'

### ISSUE THREAD

**Issue by dpackham** [cdjackson/HABmin#306]
I have running 1.8.2 with the 4 week old zwave jar 1.9.0 from the ZWave-Security-Testing wiki page.  

My Kwikset 910 locks are coming up with "NODE 12: No database entry: Manufacturer:144 [ID:1,Type:1]"

when I run OH2 it see them as Kwikset 914 locks and does not throw errors.  how can i update the OH1 jar to see the 910's?

**Comment by dpackham** [comment]
I see a file in the database.. with the Man of 144...  should it connect with that?
![untitled](https://cloud.githubusercontent.com/assets/707522/14983742/4890e114-10fc-11e6-8d3f-6767724b1b9e.png)

**Comment by cdjackson** [comment]
Are yo compiling the binary yourself?

**Comment by dpackham** [comment]
no   downloaded from a link posted UHH somewhere.  a few iterations.  where is the REAL latest one?

**Comment by cdjackson** [comment]
There is no current binding that supports security, so unfortunately we can’t add this device into a binding that has the security classes.

**Comment by dpackham** [comment]
hmmm...  i followd this https://github.com/openhab/openhab/wiki/ZWave-Security-Testing and it looks like its working... just not finding the XML files for the locks.   right?

**Comment by dpackham** [comment]
can/should I compile my own jar? to help test?

**Comment by cdjackson** [comment]
No -this version is not the released version. The current version of zwave does not contain support for locks.

**Comment by dpackham** [comment]
OK cool.   when I load up OH2 it sees the locks and configured them automagically..  is there dev code I can run or get from cloudbees?  or will the secure lock code only be in OH2?

**Comment by cdjackson** [comment]
It will likely only be in OH2. I’m not convinced it’s working well although I have used it with my Yale locks but I intend to spend some time looking at this over the next week or two.

**Comment by dpackham** [comment]
ok.   so i do see the XML file in your online database can I compile a jar and include that?  I would be willing to test with/for you.   im in the middle of migrating from my Vera ( with your UI there ) to OH so im willing to give some time right now  :)

**Comment by cdjackson** [comment]
Yes - you should be able to compile a JAR yourself and export the XML from the database.

I don’t have an OH1 environment at the moment so am really doing most of the development in OH2. The security classes aren’t 100% at the moment so I need to spend some time to resolve that and will likely spend this weekend at it.

**Comment by dpackham** [comment]
Cool... let me know if you want me to test.  I can also allow you SSH into my OH1 box for test as well.   its not critical for me right now so its good timing

**Comment by dpackham** [comment]
can I get access to add the 910 xml to your database?  i have an account under username dpackham now.  my 910 says its manufacturer

 ID 144.  prolly due to the zwave card saying black and decker on it.

Attaching photos

![img_20160501_221800](https://cloud.githubusercontent.com/assets/707522/16348853/44f2646c-3a12-11e6-8e40-f333bc30e5c0.jpg)

![untitled](https://cloud.githubusercontent.com/assets/707522/16348858/4bffce34-3a12-11e6-93a0-86abfcc13297.png)

**Comment by cdjackson** [comment]
Hi,
I’ve updated your access.

Regarding the manufacturer, this is currently set as Kwikset. Probably, this name is wrong - it’s probably an OEM manufacturer, but we can’t have multiple manufacturers with the same ID…

What is a ‘910’ anyway?

Cheers
Chris

**Comment by dpackham** [comment]
Just a deadbolt.  so if I add this can I build a secure jar to control the lock?  or do I still need to go to OH2 for the control?

OH2 doesnt see the 910 either yet  :)

Dave


![traditional-nickel](https://cloud.githubusercontent.com/assets/707522/16349028/6f1608a6-3a13-11e6-8a89-7b5bb2ae2419.png)

**Comment by dpackham** [comment]
so I added it but I think I messed up.  its really just the same as the 914 that you have already in the database (I think) with a new man ID of 144 I think due to the B&D zwave card in it...  Teach me Obi Wan  ;)

**Comment by cdjackson** [comment]
It looks like it created a new device when it should have updated the existing device. For some reason the manufacturer is showing incorrectly as 0…

I just had a look at the XML files you loaded - they aren’t the right type. The XML that should have been loaded i the one that OH creates automatically - not the database files which is what it looks like you’ve loaded…

So, I’ve deleted the new entry that you created - I think the original version is fine despite the B&D label. The manufacturer is the same as the other Kwikset 914, so it should just use that...

**Comment by dpackham** [comment]
Here is what OH created for that node...

<node>
  <deviceClass>
    <basicDeviceClass>ROUTING_SLAVE</basicDeviceClass>
    <genericDeviceClass>ENTRY_CONTROL</genericDeviceClass>
    <specificDeviceClass>SECURE_KEYPAD_DOOR_LOCK</specificDeviceClass>
  </deviceClass>
  <homeId>0xe3c4c926</homeId>
  <nodeId>25</nodeId>
  <version>3</version>
  <name>Garage Door Deadbolt</name>
  <location>Garage Door</location>
  <manufacturer>0x90</manufacturer>
  <deviceId>0x1</deviceId>
  <deviceType>0x1</deviceType>
  <listening>false</listening>
  <frequentlyListening>true</frequentlyListening>
  <routing>true</routing>
  <security>false</security>
  <beaming>true</beaming>
  <maxBaudRate>40000</maxBaudRate>
  <nodeInformationFrame/>
  <supportedCommandClasses>
    <entry>
      <commandClass>MANUFACTURER_SPECIFIC</commandClass>
      <manufacturerSpecificCommandClass>
        <version>1</version>
        <instances>1</instances>
      </manufacturerSpecificCommandClass>
    </entry>
    <entry>
      <commandClass>NO_OPERATION</commandClass>
      <noOperationCommandClass>
        <version>1</version>
        <instances>1</instances>
      </noOperationCommandClass>
    </entry>
    <entry>
      <commandClass>BASIC</commandClass>
      <basicCommandClass>
        <version>1</version>
        <instances>1</instances>
        <isGetSupported>true</isGetSupported>
      </basicCommandClass>
    </entry>
    <entry>
      <commandClass>VERSION</commandClass>
      <versionCommandClass>
        <version>1</version>
        <instances>1</instances>
        <libraryType>LIB_SLAVE_ENHANCED</libraryType>
        <protocolVersion>3.28</protocolVersion>
        <applicationVersion>2.6</applicationVersion>
      </versionCommandClass>
    </entry>
  </supportedCommandClasses>
  <nodeNeighbors>
    <int>1</int>
    <int>13</int>
    <int>18</int>
    <int>19</int>
    <int>20</int>
    <int>24</int>
    <int>27</int>
    <int>32</int>
  </nodeNeighbors>
  <lastSent>2016-06-24 19:52:25.199 UTC</lastSent>
  <lastReceived>2016-06-24 19:52:44.164 UTC</lastReceived>
</node>

**Comment by dpackham** [comment]
maybe for now ill run my locks from my vera with the openhab binding

and keep trying to get OH2 working well enough to run them

Dave P
